java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -767,4 +794,40 @@ public String getClientLibToken() {
   public String getQuotaProjectId() {
     return quotaProjectId;
   }
+
+  /** Returns the resolved endpoint for the Service to connect to Google Cloud */
+  public String getResolvedEndpoint(String serviceName) {
+    if (universeDomain == null) {
+      return formatEndpoint(serviceName, GOOGLE_DEFAULT_UNIVERSE);
+    } else if (Strings.isNullOrEmpty(universeDomain)) {
+      throw new IllegalArgumentException("Universe Domain cannot be empty");
+    } else {
+      if (host == null) {
+        return formatEndpoint(serviceName, getUniverseDomain());
+      }
+      return host;
```

Comment:
Can we refactor the logic in this method to something like
```
if (host != null) {
  return host;
}
//check universeDomain
``` 
This would be clearer to comprehend that `host` should override `universeDomain` if specified, and reduce the nested indentations.

Reply:
Sure, will update!